ckb-vm-fuzzing-utils/src/lib.rs

```diff
@@ -177,13 +177,29 @@ where
                 impls.load_input_by_field(buf, offset, index, source, field)
             })?,
             SyscallCode::LoadCellDataAsCode => {
-                let addr = machine.registers()[A0].to_u64() as *mut u8;
-                let memory_size = machine.registers()[A1].to_u64() as usize;
+                let addr = machine.registers()[A0].to_u64();
+                let memory_size = machine.registers()[A1].to_u64();
                 let content_offset = machine.registers()[A2].to_u64() as usize;
                 let content_size = machine.registers()[A3].to_u64() as usize;
                 let index = machine.registers()[A4].to_u64() as usize;
                 let source = machine.registers()[A5].to_u64().try_into().expect("parse source");
-                let result = self.impls.load_cell_code(addr, memory_size, content_offset, content_size, index, source);
+
+                let mut buf = vec![0u8; memory_size as usize];
+                let result = self.impls.load_cell_code(
+                    buf.as_mut_ptr() as *mut u8,
+                    memory_size as usize,
+                    content_offset,
+                    content_size,
+                    index,
+                    source,
+                );
+                machine.memory_mut().store_bytes(addr, &buf)?;
```

Comment:
I'm not sure about this, the whole idea of `SyscallImplsSynchronousWrapper` is that underlying `impls` should handle the full syscall. What's the storing and permission setting doing here?